packages/typespec-vscode/src/typespec-utils.ts

```diff
@@ -53,3 +37,56 @@ export async function TraverseMainTspFileInWorkspace() {
         .map((uri) => normalizeSlashes(uri.fsPath)),
     );
 }
+
+export async function TraverseEntrypointTspFileInWorkspace(): Promise<string[]> {
+  const targetFolders = await vscode.workspace
+    .findFiles(`**/${StartFileName}`, "**/node_modules/**")
+    .then((uris) => {
+      const directories = new Set<string>();
+      uris
+        .filter((uri) => uri.scheme === "file" && !uri.fsPath.includes("node_modules"))
+        .map((uri) => directories.add(getDirectoryPath(normalizeSlashes(uri.fsPath))));
+      return Array.from(directories);
+    });
+  const entrypointTspFiles: string[] = [];
+
+  for (const folder of targetFolders) {
+    while (true) {
+      const entrypointTspFile = await getEntrypointTspFileInFolder(folder);
+      if (entrypointTspFile) {
+        entrypointTspFiles.push(entrypointTspFile);
+      }
+    }
+  }
+
+  return entrypointTspFiles;
+}
+
+async function getEntrypointTspFileInFolder(folder: string): Promise<string | undefined> {
```

Comment:
you shouldn't rewrite that this is not the actual logic, you are missing `exports` which are a very complex resolution
We should share this logic from the compiler

Reply:
+1

Reply:
Agree. @timotheeguerin , maybe you can help to also expose the "resolveTypeSpecEntrypoint" too when you are separating the PR for init bundling if it's easy. thx.